bindings/gcp/bucket/metadata.yaml

```diff
@@ -0,0 +1,106 @@
+# yaml-language-server: $schema=../../../component-metadata-schema.json
+schemaVersion: v1
+type: bindings
+name: gcp.bucket
+version: v1
+status: alpha
+title: "GCP Storage Bucket"
+urls:
+  - title: Reference
+    url: https://docs.dapr.io/reference/components-reference/supported-bindings/gcpbucket/
+binding:
+  output: true
+  operations:
+    - name: create
+      description: "Create an item."
+capabilities: []
+authenticationProfiles:
+  - title: "GCP API call Authentication"
+    description: |
+      Authenticate authenticates API calls with the given service account or refresh token JSON credentials.
+    metadata:
+      - name: private_key_id
+        required: true
+        sensitive: true
+        description: |
+          GCP private key id.
+        example: '"privateKeyId"'
+      - name: private_key
+        required: true
+        sensitive: true
+        description: |
+          GCP credentials private key. Replace with x509 cert.
+        example: '"12345-12345"'
```

Comment:
Example should be an actual private key?

Also, the name is "private key" but then this refers to X.509 certificates. Can you confirm if this is a certificate or private key? And how should it be encoded, as PEM?

Reply:
@ItalyPaleAle This appears to be following the naming convention from the GCP Service Account Key JSON. The `private_key` is a X.509 certificate.  Here is a sample:
```
{
  "type": "service_account",
  "project_id": "someproject-123456",
  "private_key_id": "4d5ac58cc2b5a01e23223123341284",
  "private_key": "-----BEGIN PRIVATE KEY-----\nMIIEvgIBADANBgkqhkiG9w0BAQEv5vu8sUw+2/y/MfhhkIKQTgnoiIZFHximeQN67VF9pifNR/D0k\nA9W9wIr1sJJ4LITvgiui0GqzsbvW98KqeuXll1qJVlVHVSp64gMEYVSVZAK47vNs\nVP2HbvhpWdhBcKRFHs06BPyD\n-----END PRIVATE KEY-----\n",
  "client_email": "someone@someproject-123456.iam.gserviceaccount.com",
  "client_id": "1234567890",
  "auth_uri": "https://accounts.google.com/o/oauth2/auth",
  "token_uri": "https://oauth2.googleapis.com/token",
  "auth_provider_x509_cert_url": "https://www.googleapis.com/oauth2/v1/certs",
  "client_x509_cert_url": "https://www.googleapis.com/robot/v1/metadata/x509/someproject-123456.iam.gserviceaccount.com"
}
```
I can use the above `private_key` as the example provided in the Metadata schema.

Reply:
This is not a X.509 certificate? I see a JSON payload.

As for using this as example (truncated, for example we don't need to see all values for all properties), sounds good

Reply:
You're right it not a X.509.  I changed the description.